bin/check-subnet-ip-consumption.rb

```diff
@@ -0,0 +1,246 @@
+#! /usr/bin/env ruby
+#
+# check-subnet-ip-consumption
+#
+#
+# DESCRIPTION:
+#   This plugin uses the EC2 API to determine if any subnet in a given region
+#   has consumed IP addresses such that the consumption exceeds a user-specified threshold.
+#   This plugin additionally uses the IAM API to resolve the account alias,
+#   if the user uses the -s/--show-account-alias flag.
+#
+# OUTPUT:
+#   plain-text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: aws-sdk
+#   gem: sensu-plugin
+#   gem: sensu-plugins-aws
+#
+# USAGE:
+#  ./check-subnet-ip-consumption.rb -a <access key> -k <secret key> -r <region> -t <integer percentage>
+#
+# NOTES:
+#   Special thanks to Garrett Kuchta and Antonio Beyah for their assistance.
+#
+# LICENSE:
+#   Nick Jacques <Nick.Jacques@target.com>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+
+require 'sensu-plugin/check/cli'
+require 'sensu-plugins-aws'
+require 'aws-sdk'
+
+class CheckSubnetIpConsumption < Sensu::Plugin::Check::CLI
+  include Common
+
+  option :aws_access_key,
```

Comment:
preference is to not add these for new checks as they should be passed in via the environment or another mechanism.  Passing in the keys on the command line is a bad idea.